flask/views.py

```diff
@@ -117,6 +117,9 @@ def __init__(cls, name, bases, d):
 
         if 'methods' not in d:
             methods = set()
+            for base in bases:
+                if base.__dict__.get('methods'):
```

Comment:
should we use `hasattr()` instead?

Reply:
I use it to check both the existence of the `method` variable and whether it's empty. Do you think it makes sense?

Reply:
`getattr(base, 'methods', None)` would do the same

Reply:
Yes, that does make sense! Thanks for the clarification! I've fixed it.